great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -348,8 +348,9 @@ def __init__(
             )
         self._ge_cloud_mode = ge_cloud_mode
         self._ge_cloud_config = ge_cloud_config
-        self._project_config = project_config
-        self._apply_global_config_overrides()
+        self._project_config = self._apply_global_config_overrides(
+            config=project_config
+        )
```

Comment:
The critical change. We intended this to have no side-effect, but the tests have revealed that trying to modify the config in place has resulted in some behavior that was hidden. Further investigation is needed as to the root cause